File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position :: position) :: integer
  def score({x, y}) do
    dist = :math.sqrt(x * x + y * y)
    cond do
      dist <=  1.0 -> 10
      dist <=  5.0 ->  5
      dist <= 10.0 ->  1
      true         ->  0
    end
  end
end

File: lua/darts/darts.lua
local Darts = {}

local function hypot(x, y)
    return math.sqrt(x^2 + y^2)
end


function Darts.score(x, y)
    local dist = hypot(x, y)
    if dist <=  1 then return 10 end
    if dist <=  5 then return  5 end
    if dist <= 10 then return  1 end
    return 0
end

return Darts
